### PR TITLE
Factor out DescriptionPackage's path-related functions to PackageLocator protocol

### PR DESCRIPTION
**Pull request by ikesyo**
This would make it easy for testing FrameworkComponentsCollector and FrameworkModuleMapGenerator.

### CODE REVIEW

File: Sources/ScipioKit/Producer/PIF/XCBuildClient.swift [hunk 1]
```diff
@@ -4,25 +4,25 @@ import PackageGraph
...
     private let executor: any Executor
 
     init(
-        package: DescriptionPackage,
         buildProduct: BuildProduct,
         buildOptions: BuildOptions,
         configuration: BuildConfiguration,
+        packageLocator: some PackageLocator,
         fileSystem: any FileSystem = localFileSystem,
         executor: any Executor = ProcessExecutor(decoder: StandardOutputDecoder())
```
Comment (giginet) [inline Sources/ScipioKit/Producer/PIF/XCBuildClient.swift:20]: They also should be `some`

**Review by giginet — APPROVED** [review]

File: Sources/ScipioKit/Producer/PIF/XCBuildClient.swift (hunk 1 above)
Comment (ikesyo) [reply Sources/ScipioKit/Producer/PIF/XCBuildClient.swift:20]: For `localFileSystem`, that is TSCBasic's global variable and that is typed as `any FileSystem` so cannot be changed. I'll fix `executor` part.

Comment (ikesyo) [reply Sources/ScipioKit/Producer/PIF/XCBuildClient.swift]: Addressed in 67a3cad

**Comment by ikesyo** [comment]
🚢

**Pull request merged**